src/main/java/greencity/service/impl/SpecificationServiceImpl.java

```diff
@@ -83,6 +83,10 @@ public Long deleteById(Long id) {
      */
     @Override
     public Specification findByName(String name) {
+        Specification specification = specificationRepo.findByName(name);
+        if (specification == null) {
+            throw new NotFoundException(ErrorMessage.SPECIFICATION_NOT_FOUND_BY_NAME + name);
+        }
         return specificationRepo.findByName(name);
```

Comment:
you make 2 DB calls. You should return "specification" variable, but not do second request to DB.